### PR TITLE
Update redis health checks based on dot-com

### PR DESCRIPTION
**Pull request by daxmc99**
From #inc-270

### Checklist

- [ ] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [ ] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md)
- [ ] Update [Kubernetes update doc](https://docs.sourcegraph.com/admin/updates/kubernetes)

### Test plan

We should try to match the config on dot-com and keep our learnings

### CODE REVIEW

**Comment by daxmc99** [comment]
matches https://github.com/sourcegraph/deploy-sourcegraph-cloud/pull/18416

**Review by jdpleiness — APPROVED** [review]

File: charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml [hunk 1]
```diff
@@ -53,9 +53,27 @@ spec:
...
+            - -c
+            - |
+              #!/bin/bash
+              response=$(
+                redis-cli ping
+              )
+              if [ "$response" != "PONG" ]; then
+                echo "$response"
+                exit 1
+              fi
```
Comment (eseliger) [inline charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml:76]: @ggilmore made a suggestion on the call yesterday after you left that it might be nice to add this script to the image itself in sg/sg, and we only need to invoke it from here. 
Benefits:
- Portable to all deployment types, without the need to reimplement this every time, and fix it everywhere, should there be more edge-cases we didn't cover
- We can include it in our image tests to make sure the script keeps working

Comment (willdollman) [reply charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml:76]: Implemented in https://github.com/sourcegraph/sourcegraph/pull/60365

**Comment by eseliger** [comment]
No that’s fine, it’s always been like that

File: charts/sourcegraph/templates/redis/redis-store.Deployment.yaml [hunk 2]
```diff
@@ -58,16 +58,30 @@ spec:
...
+          periodSeconds: 30
+          failureThreshold: 2
+          timeoutSeconds: 5
           tcpSocket:
             port: redis
+        readinessProbe:
+          initialDelaySeconds: 5
+          exec:
+            command:
+            - /bin/sh
```
[the remark this answers is not in the document]
Comment (eseliger) [reply charts/sourcegraph/templates/redis/redis-store.Deployment.yaml:71]: @daxmc99 any chance you can make this change?

otherwise I’ll try to give it a go at some point

**Comment by keegancsmith** [comment]
@eseliger chatted with dax slack and he ended up following up and wants to cherry-pick the final state in (ie use the readiness script): https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/420

Look good to you?

**Comment by keegancsmith** [comment]
@daxmc99 I think given the above, I'm more comfortable leaving this out of 5.3

**Comment by daxmc99** [comment]
@eseliger @keegancsmith 
Agreed, no backport to 5.3 here
